tests/baselines/reference/contextualTypeAny.errors.txt

```diff
@@ -0,0 +1,15 @@
+tests/cases/compiler/contextualTypeAny.ts(3,5): error TS2322: Type '{ p: string; q: any; }' is not assignable to type '{ [s: string]: number; }'.
+  Property 'p' is incompatible with index signature.
+    Type 'string' is not assignable to type 'number'.
+
+
+==== tests/cases/compiler/contextualTypeAny.ts (1 errors) ====
+    var x: any;
+    
+    var obj: { [s: string]: number } = { p: "", q: x };
+        ~~~
+!!! error TS2322: Type '{ p: string; q: any; }' is not assignable to type '{ [s: string]: number; }'.
+!!! error TS2322:   Property 'p' is incompatible with index signature.
+!!! error TS2322:     Type 'string' is not assignable to type 'number'.
+    
+    var arr: number[] = ["", x];
```

Comment:
Yeah, would be nice to catch this, but it would require adding discrete properties 0, 1, 2,... to the array which I don't think anyone wants. Or perhaps we could consider giving array literals type `any[]` only when all elements are of type `any` and otherwise just ignore the `any` elements. Basically like saying that `any` elements in an array literal are presumed to be of the same type(s) as the non-`any` elements. Not sure how well that would work, though.